[Cocoa] WPT Tests referencing "white.webm" render video as non-white

jernoble opened the pull request:
#### 7e678d8d2dc8c74b2727beba6e8dc9e843edbcb5
<pre>
[Cocoa] WPT Tests referencing &quot;white.webm&quot; render video as non-white
<a href="https://bugs.webkit.org/show_bug.cgi?id=278246">https://bugs.webkit.org/show_bug.cgi?id=278246</a>
<a href="https://rdar.apple.com/134075823">rdar://134075823</a>

Reviewed by Eric Carlson.

In the absence of explicit color range information (from the WebM &quot;Colour&quot; element),
default the VP8 decoder to &quot;limited range&quot; (16-235) color values. This matches the
behavior of Firefox and Chrome.

* LayoutTests/platform/mac/TestExpectations:
* Source/WebCore/platform/graphics/cocoa/VP9UtilitiesCocoa.mm:
(WebCore::createVideoInfoFromVP8Header):

Canonical link: <a href="https://commits.webkit.org/282441@main">https://commits.webkit.org/282441@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/410e059cb5f41c4c270b3353ac6df58ec832d8e5

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/63161 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/42517 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/15757 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/67182 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/13769 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/65281 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/50204 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/14049 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/50900 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/9510 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/66230 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/39484 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/54690 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/31586 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/36169 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/12037 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/12641 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/57709 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/12366 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/68877 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/7107 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/11988 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/58214 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/7139 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/54762 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/58429 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13999 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/5930 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/38337 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/39417 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/40528 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/39159 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/platform/mac/TestExpectations:
```diff
@@ -863,6 +863,40 @@ imported/w3c/web-platform-tests/media-source/mediasource-trackdefaultlist.html [
+# --- Start WebVTT Tests ---
+## --- Start W3C Imported WebVTT Tests ---
+
+# imported/w3c/web-platform-tests/webvtt/rendering/cues-with-video tests are marked as [ ImageOnlyFailure ] by default
+# Add newly passing tests to this file as Apple-port-specific bugs are fixed:
```
(potentially a separate issue) but inverting this would be better, so new tests that are imported get run

danae404 replied:
There are about 250 tests in this directory - inverting this so that over 200 tests are marked individually as failing could possibly impact readability of this file (but this probably isn't relevant to this patch)

jernoble committed:
[Cocoa] WPT Tests referencing "white.webm" render video as non-white
https://bugs.webkit.org/show_bug.cgi?id=278246
rdar://134075823

Reviewed by Eric Carlson.

In the absence of explicit color range information (from the WebM "Colour" element),
default the VP8 decoder to "limited range" (16-235) color values. This matches the
behavior of Firefox and Chrome.

* LayoutTests/platform/mac/TestExpectations:
* Source/WebCore/platform/graphics/cocoa/VP9UtilitiesCocoa.mm:
(WebCore::createVideoInfoFromVP8Header):

Canonical link: https://commits.webkit.org/282441@main
